### DISCUSSION TITLE
[BUG] Wireguard: Configuration is invalid

### DISCUSSION THREAD

**Discussion by yoobi** [pivpn/pivpn#1382] [Support]
## In raising this issue, I confirm the following: 

- [x] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting *is* directly related to the pivpn installer script.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

**Describe the bug**
I had pivpn before on this raspberry, I've done `pivpn -u` and then reinstall it but when I do `pivpn -a` I get the following configuration file for Wireguard
```
[Interface]
PrivateKey = [PrivateKey]
Address = 10.6.0.2/24
MTU = 1420
DNS = 9.9.9.9, 149.112.112.112

[Peer]
PublicKey = vPii0dWIRL5YclcWXh2eSgwTbdt12MeHhbE+832/eXw=
PresharedKey = [PassKey]
Endpoint = [Endpoint]:51820
AllowedIPs = 0.0.0.0/0, ::0/0
```

And when I install it on WireGuard I get the following error : `Unable to create tunnel - The configuration is invalid`

**Expected behavior**
A correct configuration file

**Desktop (please complete the following information):**
 - VPN Protocol: Wireguard
 - OS: Raspbian
 - Version: 10 (buster)

**Comment by orazioedoardo** [comment]
> I get the following configuration file for Wireguard

Looks fine to me

> And when I install it on WireGuard I get the following error : Unable to create tunnel - The configuration is invalid

Which WireGuard client? OS?

**Comment by yoobi** [comment]
> Looks fine to me

Yeah I felt that the configuration was OK as well but I'm not too experienced with it

> Which WireGuard client? OS?
I'm on macOS Mojave (10.14.6) and with the official WireGuard client from appstore

**Comment by FlyingCodeMonkey** [comment]
Did you ever get to the bottom of this? I'm having the same issue with the official WireGuard client from the AppStore on macOS. Same error "Unable to create tunnel - The configuration is invalid"
Looking at the log files, I see:

```
2022-04-27 11:10:56.900 [APP] Unable to add config to keychain: -60005
2022-04-27 11:10:56.905 [APP] Add: Saving configuration failed: Error Domain=NEVPNErrorDomain Code=1 "Missing protocol or protocol has invalid type" UserInfo={NSLocalizedDescription=Missing protocol or protocol has invalid type}
```

Any thoughts?

**Reply by yoobi** [reply]
I'm sorry I don't remember how I fixed this. I'm currently using openvpn on my NAS instead

**Reply by Abduladzhi** [reply]
Исправили?
